src/Indigo/Contracts/HomebaseLite/Types.hs

```diff
@@ -0,0 +1,67 @@
+-- SPDX-FileCopyrightText: 2022 Tezos Commons
+-- SPDX-License-Identifier: LicenseRef-MIT-TC
+
+{-# OPTIONS -Wno-orphans #-}
+
+module Indigo.Contracts.HomebaseLite.Types
+  ( Parameter(..)
+  , Storage(..)
+  , Configuration(..)
+  , URI(..)
+  ) where
+
+import Indigo
+
+data Parameter
+  = Set_admin KeyHash
+  | Accept_admin ()
+  | Add_maintainers [KeyHash]
+  | Remove_maintainers [KeyHash]
+  | Configure Configuration
+  | Propose (URI, [MText])
+  | Vote (URI, Natural)
+  deriving stock (Generic, Show)
+  deriving anyclass (IsoValue)
+
+[entrypointDoc| Parameter plain |]
+
+data Configuration = Configuration
+  { cExpireTime :: Natural
+  , cVoteDelay :: Natural
+  , cQuorumThreshold :: Natural
+  , cMinimumBalance :: Natural
+  }
+  deriving stock (Generic, Show)
+  deriving anyclass (IsoValue, HasAnnotation)
+
+[typeDoc| Configuration "Options defining the behaviour and life-cycle of proposals."|]
+
+data ProposalInfo = ProposalInfo
+  { piConfiguration :: Configuration
+  , piLevel :: Natural
+  , piTime :: Timestamp
+  , piChoices :: [MText]
+  }
+  deriving stock (Generic, Show)
+  deriving anyclass (IsoValue, HasAnnotation)
+
+[typeDoc| ProposalInfo "Information defining a proposal."|]
+
+newtype URI = URI MText
+  deriving stock (Generic, Show, Eq, Ord)
+  deriving anyclass (IsoValue, HasAnnotation)
+
+[typeDoc| URI "Text representing IPFS URI for a proposal."|]
+
+data Storage = Storage
+  { sAdmin :: KeyHash
+  , sAdminCandidate :: Maybe KeyHash
```

Comment:
A common trick for efficiency here (that we used in other contracts) is to store this without the `Maybe`.

The case in which there is no candidate is stored as if the candidate is the current `admin`.

This reduces the size of the stored data (admittedly very little), but also simplifies the implementation, because it removes the need to wrap/unwrap the `address` as well as some other check that would need to happen, while still fitting in the logic described in the spec.